src/codegen.rs

```diff
@@ -503,7 +504,486 @@ fn codegen_recursive_helper(
         assert_eq!(expr[id].nodes.len(), 1);
         &expr[id].nodes[0]
     } {
-        Language::RelayOperatorCall(_) => todo!(),
+        Language::RelayOperatorCall(ids) => {
+            let relay_op = match &expr[ids[0]].data {
+                MyAnalysisData::RelayOperator(op) => op,
+                _ => panic!(),
+            };
+
+            match relay_op {
+                RelayOperator::RelayBatchNormInference => {
+                    let data = codegen_recursive_helper(
+                        expr,
+                        ids[1],
+                        top_level_id,
+                        uninitialized_allocations_prefix,
+                        declarations,
+                        code,
+                        hw_map,
+                    );
+                    let gamma = codegen_recursive_helper(
+                        expr,
+                        ids[2],
+                        top_level_id,
+                        uninitialized_allocations_prefix,
+                        declarations,
+                        code,
+                        hw_map,
+                    );
+                    let beta = codegen_recursive_helper(
+                        expr,
+                        ids[3],
+                        top_level_id,
+                        uninitialized_allocations_prefix,
+                        declarations,
+                        code,
+                        hw_map,
+                    );
+
+                    let moving_mean = codegen_recursive_helper(
+                        expr,
+                        ids[4],
+                        top_level_id,
+                        uninitialized_allocations_prefix,
+                        declarations,
+                        code,
+                        hw_map,
+                    );
+
+                    let moving_var = codegen_recursive_helper(
+                        expr,
+                        ids[5],
+                        top_level_id,
+                        uninitialized_allocations_prefix,
+                        declarations,
+                        code,
+                        hw_map,
+                    );
+
+                    let axis = MyAnalysis::get_usize(ids[6], expr);
+
+                    // Pre-ISCA: expect NHWC format data
+                    assert!(axis == 3, "expected NHWC format");
+                    let epsilon = match &expr[ids[7]].data {
+                        MyAnalysisData::Literal(l) => {
+                            println!("shape of batchnorm inference epsilon: {:?}", &l.shape());
```

Comment:
```suggestion
```
I assume we don't need this anymore? Can you check for other leftover print statements?